codespell_lib/_codespell.py

```diff
@@ -188,10 +187,10 @@ def open_with_internal(self, filename):
                         pass
 
                 curr += 1
-
-            finally:
+            else:
+                lines = f.readlines()
                 f.close()
```

Comment:
The only line in the try clause creates the variable f, so if it fails there is no variable f and thus nothing to close